src/EditorFeatures/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs

```diff
@@ -7876,245 +8134,334 @@ abstract class Class : IInterface
 {
     public abstract void Method1();
 }",
-index: 1);
+                Options = { AllOptionsOff },
+                CodeActionEquivalenceKey = "False;True;True:global::IInterface;TestProject;Microsoft.CodeAnalysis.ImplementInterface.AbstractImplementInterfaceService+ImplementInterfaceCodeAction;",
+                CodeActionIndex = 1,
+            }.RunAsync();
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
         public async Task TestInaccessibleMember_01()
         {
-            await TestWithAllCodeStyleOptionsOffAsync(
-@"
-<Workspace>
-    <Project Language=""C#"" AssemblyName=""Assembly1"" CommonReferences=""true"">
-        <Document>
-public interface IInterface
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"class Class : {|CS0535:{|CS0535:{|CS0535:IInterface|}|}|}
+{
+}",
+                    },
+                    AdditionalProjects =
+                    {
+                        ["Assembly1"] =
+                        {
+                            Sources =
+                            {
+                                @"public interface IInterface
 {
     void Method1();
 
     internal void M1();
     internal int P1 {get;}
-}
-        </Document>
-    </Project>
-    <Project Language=""C#"" AssemblyName=""Assembly2"" CommonReferences=""true"">
-        <ProjectReference>Assembly1</ProjectReference>
-        <Document>
-class Class : [|IInterface|]
-{
-}
-        </Document>
-    </Project>
-</Workspace>",
-@"
-class Class : IInterface
+}",
+                            },
+                        },
+                    },
+                    AdditionalProjectReferences = { "Assembly1" },
+                },
+                FixedState =
+                {
+                    Sources =
+                    {
+                        @"class Class : {|CS0535:{|CS0535:IInterface|}|}
 {
     public void Method1()
     {
         throw new System.NotImplementedException();
     }
-}
-        ");
+}",
+                    },
+                    MarkupHandling = MarkupMode.Allow,
+                },
+                Options = { AllOptionsOff },
+
+                // Specify the code action by equivalence key only to avoid trying to implement the interface explicitly with a second code fix pass.
+                CodeActionEquivalenceKey = "False;False;True:global::IInterface;Assembly1;Microsoft.CodeAnalysis.ImplementInterface.AbstractImplementInterfaceService+ImplementInterfaceCodeAction;",
```

Comment:
📝 The test library could be improved for this case (only require matching key/index if the key appears somewhere in the list of available code actions, plus require the first pass have a matching pair). I'll file a follow-up issue in dotnet/roslyn-sdk.

Reply:
➡️ https://github.com/dotnet/roslyn-sdk/issues/874